### PR TITLE
Add lodash.spread

### PR DESCRIPTION
**Pull request by naholyr**
Here is the PR to implement #756 's `_.spread`

### CODE REVIEW

**Comment by jdalton** [comment]
Have you signed [the CLA](http://dojofoundation.org/about/claForm)?

**Comment by naholyr** [comment]
Had not, done now!

File: lodash.src.js [hunk 1]
```diff
@@ -7499,6 +7499,35 @@
     }
 
     /**
+     * Creates a function accepting an array as argument, that invokes `func`
+     * with arguments taken from this array, like `Function#apply`.
+     * This works obviously well with `Promise.all`.
```
Comment (jdalton) [inline lodash.src.js]: Minor doc nit. Can you provide a usage example with `Promise.all`.

File: lodash.src.js [hunk 2]
```diff
@@ -7499,6 +7499,35 @@
...
+     * @example
+     *
+     * var spread = _.spread(function (who, what) {
+     *   return who + ' says ' + what;
+     * });
+     *
+     * spread(['John', 'hello']) // => 'John says hello'
+     */
+    function spread (func, thisArg) {
+      if (!isFunction(func)) {
```
Comment (jdalton) [inline lodash.src.js]: Also can you make this `typeof func != 'function'`. It's be consistent with some local commits I've done.

**Comment by jdalton** [comment]
So close, ping.

**Comment by naholyr** [comment]
Let me know what you think of later changes

**Comment by jdalton** [comment]
I'll take it! Thanks!

**Pull request merged**